libbeat/asset/registry.go

```diff
@@ -22,36 +22,66 @@ import (
 	"compress/zlib"
 	"encoding/base64"
 	"io/ioutil"
+	"sort"
 )
 
 // FieldsRegistry contains a list of fields.yml files
 // As each entry is an array of bytes multiple fields.yml can be added under one path.
 // This can become useful as we don't have to generate anymore the fields.yml but can
 // package each local fields.yml from things like processors.
-var FieldsRegistry = map[string]map[string]string{}
+var FieldsRegistry = map[string]map[int]map[string]string{}
 
 // SetFields sets the fields for a given beat and asset name
-func SetFields(beat, name string, asset func() string) error {
+func SetFields(beat, name string, p Priority, asset func() string) error {
 	data := asset()
 
+	priority := int(p)
+
 	if _, ok := FieldsRegistry[beat]; !ok {
-		FieldsRegistry[beat] = map[string]string{}
+		FieldsRegistry[beat] = map[int]map[string]string{}
+	}
+
+	if _, ok := FieldsRegistry[beat][priority]; !ok {
+		FieldsRegistry[beat][priority] = map[string]string{}
 	}
-	FieldsRegistry[beat][name] = data
+
+	FieldsRegistry[beat][priority][name] = data
 
 	return nil
 }
 
 // GetFields returns a byte array contains all fields for the given beat
 func GetFields(beat string) ([]byte, error) {
 	var fields []byte
-	for _, data := range FieldsRegistry[beat] {
-		output, err := DecodeData(data)
-		if err != nil {
-			return nil, err
+
+	// Get all priorities and sort them
+	beatRegistry := FieldsRegistry[beat]
+	priorities := make([]int, 0, len(FieldsRegistry))
```

Comment:
Sorry for being late, but shouldn't this be `len(beatRegistry)`?

Reply:
@simitt you are right of course. Will open a PR with a fix.

Reply:
I opened https://github.com/elastic/beats/pull/9986